### ISSUE TITLE
AttributeError: 'TimeSeriesKMeans' object has no attribute 'gamma_sdtw'`

### ISSUE THREAD

**Issue by jscheithe** [rtavenar/tslearn#19]
First of all, thank you so much for sharing this wonderful package! 
You're making my master's thesis much more enjoyable :)

I'm getting a weird error and haven't been able to figure it out. Here's a minimal working example (I can call `TimeSeriesKMeans(metric='softdtw').fit(X)` without `Pipeline()` without problems).

I'm on p36, Windows 7 x64. Thanks in advance!

```
import numpy as np
from sklearn.pipeline import Pipeline
from tslearn.clustering import TimeSeriesKMeans
from tslearn.utils import to_time_series_dataset

X = to_time_series_dataset(np.array([1,2]))
pipeline = Pipeline([
        ('kmeans', TimeSeriesKMeans())
        ])
parameters = {
        'kmeans__metric' : 'softdtw'
        }
pipeline.set_params(**parameters)
pipeline.fit(X)
```

``` 
Traceback (most recent call last):

  File "<ipython-input-8-c4ba89694fd5>", line 10, in <module>
    pipeline.fit(X)

  File "C:\...\sklearn\pipeline.py", line 250, in fit
    self._final_estimator.fit(Xt, y, **fit_params)

  File "C:\...\tslearn\clustering.py", line 531, in fit
    self._fit_one_init(X_, x_squared_norms, rs)

  File "C:\...\tslearn\clustering.py", line 460, in _fit_one_init
    self._assign(X)

  File "C:\...\tslearn\clustering.py", line 480, in _assign
    dists = cdist_soft_dtw(X, self.cluster_centers_, gamma=self.gamma_sdtw)

AttributeError: 'TimeSeriesKMeans' object has no attribute 'gamma_sdtw'
```

**Comment by jscheithe** [comment]
The problem always occurs when changing the parameter "metric" to "softdtw" after creating an instance of TimeSeriesKMeans() with a different metric.

I'm not sure how this is handled in `sklearn` in general, so this probably is not a bug, but maybe it's safer to allow it. An easy fix would be deleting line 450 from classifiers.py:

`if self.metric == "softdtw":`

to always set (regardless of the metric paramter)

`self.gamma_sdtw = metric_params.get("gamma_sdtw", 1.)`

**Comment by rtavenar** [comment]
Hi,

Thanks for the bug report and for the fix! :)

It's implemented now (on github and in version 0.1.10.7).

**Comment by jscheithe** [comment]
Thanks for committing the fix, glad I could help!
